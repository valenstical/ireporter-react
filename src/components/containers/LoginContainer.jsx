import React, { Component } from 'react';
import { connect } from 'react-redux';
import formDataJSON from 'formdata-json';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { Redirect } from 'react-router-dom';
import Login from '../views/Login';
import { getUser } from '../../selectors';
import authenticate from '../../actions/authAction';
import { scrollTop } from '../../utils/helpers';


/**
 * Container component for the Login view
 * @export
 * @class LoginContainer
 * @extends {Component}
 */
class LoginContainer extends Component {
  handleLogin = async (event) => {
    event.preventDefault();
    const payload = formDataJSON(new FormData(event.target));
    payload.route = 'auth/login';
    payload.method = 'post';
    const { login: processLogin } = this.props;
    await processLogin(payload);
    scrollTop();
  }

  render() {
    const { user } = this.props;
    if (user.isLoggedIn && !user.isAdmin) {
      return <Redirect to="/dashboard" />;
    }
    if (user.isLoggedIn && user.isAdmin) {
      return <Redirect to="/admin" />;
    }
    return (
      <Login
      handleSubmit={this.handleLogin}
      user={user}
      />
    );
  }
}
LoginContainer.propTypes = {
  user: PropTypes.object.isRequired,
  login: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector(
  {
    user: getUser,
  }
);

const mapDispatchToProps = dispatch => ({
  login: payload => dispatch(authenticate(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);
