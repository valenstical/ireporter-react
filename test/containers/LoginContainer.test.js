import React from 'react';
import setup from '../setup';
import LoginContainer from '../../src/components/containers/LoginContainer';


describe('<LoginContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(<LoginContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
