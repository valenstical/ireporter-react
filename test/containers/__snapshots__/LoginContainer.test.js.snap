// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginContainer)>
        <LoginContainer
          isBusy={false}
          login={[Function]}
          message={Array []}
          success={false}
        >
          <Login
            handleSubmit={[Function]}
            isBusy={false}
            message={Array []}
            success={false}
          >
            <section
              className="how login-wrapper bg-grey top-space"
              data-pg-name="How it Works"
            >
              <div
                className="wrapper"
                data-pg-name="Wrapper"
              >
                <form
                  className="card login-form"
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <header
                    className="form-header"
                  >
                    <h1
                      className="text-brand"
                    >
                      Welcome Back
                    </h1>
                    <p>
                      Sign in to start reporting...
                    </p>
                  </header>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username, Email or Phone number
                    </label>
                    <input
                      className="form-element"
                      id="username"
                      name="username"
                      required={true}
                      type="text"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="forgotPassword"
                    >
                      Password
                    </label>
                    <Link
                      className="forgot-password transition"
                      replace={false}
                      to="#"
                    >
                      <a
                        className="forgot-password transition"
                        href="/"
                        onClick={[Function]}
                      >
                        Forgot Password?
                      </a>
                    </Link>
                    <input
                      autoComplete="off"
                      className="form-element"
                      id="forgotPassword"
                      name="password"
                      required={true}
                      type="password"
                    />
                  </div>
                  <button
                    className="btn-brand btn-submit"
                    disabled={false}
                    type="submit"
                  >
                    <span>
                      Sign In 
                    </span>
                  </button>
                </form>
                <p
                  className="login-register-link"
                >
                  New to iReporter?
                   
                  <Link
                    replace={false}
                    to="/sign-up"
                  >
                    <a
                      href="/sign-up"
                      onClick={[Function]}
                    >
                      Register here
                    </a>
                  </Link>
                </p>
              </div>
            </section>
          </Login>
        </LoginContainer>
      </Connect(LoginContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
